src/backend/joanie/payment/backends/base.py

```diff
@@ -37,21 +38,33 @@ def _do_on_payment_success(cls, order, payment):
         then mark the order as validated
         """
         # - Create an invoice
-        recipient_name = (
-            f"{payment['billing_address']['first_name']} "
-            f"{payment['billing_address']['last_name']}"
-        )
-        recipient_address = (
-            f"{payment['billing_address']['address']}\n"
-            f"{payment['billing_address']['postcode']} {payment['billing_address']['city']}, "
-            f"{payment['billing_address']['country']}"
-        )
+        try:
+            address = Address.objects.get(
+                first_name=payment["billing_address"]["first_name"],
+                last_name=payment["billing_address"]["last_name"],
+                address=payment["billing_address"]["address"],
+                postcode=payment["billing_address"]["postcode"],
+                city=payment["billing_address"]["city"],
+                country=payment["billing_address"]["country"],
+                owner=order.owner,
+            )
```

Comment:
An if it's a non reusable one, should we use it or not ?

Reply:
From my point of view if all is matching yes, in this way we prevent to have duplicated entry in the database;